x-pack/plugins/enterprise_search/public/applications/app_search/utils/role/get_role_abilities.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Account } from '../../types';
+
+import { RoleTypes, AbilityTypes, Role } from './types';
+
+/**
+ * Transforms the `role` data we receive from the Enterprise Search
+ * server into a more convenient format for front-end use
+ */
+export const getRoleAbilities = (role: Account['role']): Role => {
```

Comment:
This fn is the exact same as before, I just moved the type definitions out into `types.ts`. The types are also the same as before